app/src/main/java/io/neurolab/tools/Utils.java

```diff
@@ -0,0 +1,40 @@
+package io.neurolab.tools;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+public class Utils {
+
+    private static InputStream is;
+
+    public static int countLines(String filename) throws Exception {
+        return countLines(new File(filename));
+    }
+
+    public static int countLines(File file) throws Exception {
+
+        try (InputStream input = new BufferedInputStream(new FileInputStream(file))) {
+            is = input;
+            byte[] c = new byte[1024];
+            int count = 0;
+            int readChars = 0;
+            boolean empty = true;
+            while ((readChars = input.read(c)) != -1) {
+                empty = false;
+                for (int i = 0; i < readChars; ++i) {
+                    if (c[i] == '\n') {
+                        ++count;
+                    }
+                }
+            }
+            return (count == 0 && !empty) ? 1 : count;
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            is.close();
+        }
```

Comment:
No need for this now since you are using try with resources

Reply:
@iamareebjamal then now the  InputStream object can be closed in the try block only right?